Simulation/DetSimAna/src/Edm4hepWriterAnaElemTool.cpp

```diff
@@ -87,14 +88,17 @@ Edm4hepWriterAnaElemTool::BeginOfEventAction(const G4Event* anEvent) {
     // reset
     m_track2primary.clear();
 
+    auto SimPIonCol =  m_SimPrimaryIonizationCol.createAndPut();
```

Comment:
Hi @wenxingfang , thank you for your update. My last question: do you think that we can also move this part to the `TrackHeedSimTool`? Because I see you had to `const_cast` the collection. If you create it in the same tool, then I think you don't need to cast.

Reply:
Hi @mirguest , I don't know if there is a way to judge if this G4 simulation step is the beginning of an event in TrackHeedSimTool,  because m_SimPrimaryIonizationCol.createAndPut(); should be done at the beginning of an event. Do you have any idea?